src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -823,6 +832,28 @@ private void WriteProjectItems(IEnumerable items)
             }
         }
 
+        private const string EmbedInBinlogItemType = "EmbedInBinlog";
+
+        private void CheckForFilesToEmbed(string itemType, object itemList)
+        {
+            if (!string.Equals(itemType, EmbedInBinlogItemType, StringComparison.OrdinalIgnoreCase) || itemList is not IEnumerable list)
+            {
+                return;
+            }
+
+            foreach (var item in list)
```

Comment:
I assume that each project in the graph will add the same files over and over (because I assume the EmbedInBinlog item will be added either in directory.build.props or the sdk itself). Is that alright? Do you need to dedupe?

Reply:
it’s deduped, there’s a hashset